omnipaxos/src/ballot_leader_election.rs

```diff
@@ -138,19 +140,16 @@ impl BallotLeaderElection {
     }*/
 
     fn check_leader(&mut self) -> Option<Ballot> {
-        self.quorum_connected = true;
         let ballots = std::mem::take(&mut self.ballots);
         let top_ballot = ballots
             .into_iter()
-            .filter_map(
-                |(ballot, candidate)| {
-                    if candidate {
-                        Some(ballot)
-                    } else {
-                        None
-                    }
-                },
-            )
+            .filter_map(|(ballot, connectivity)| {
+                if self.quorum.is_accept_quorum(connectivity) {
+                    Some(ballot)
+                } else {
+                    None
+                }
+            })
```

Comment:
Since we now have the actual connectivity, it would be nice to also prioritize the nodes that have better connectivity. Right now, the `max()` function will compare the ballots lexicographically, but since connectivity is not part of the `Ballot` we probably need to map a Ballot `b`  into a tuple `(b.n, b.priority, connectivity, b.pid)` before picking out the max.